### DISCUSSION TITLE
[RFC] Start user service using C#

### DISCUSSION THREAD

**Discussion by rizaramadan** [TokoBapak/.github#11] [Backend Team Posts]
I'm planning to start implement user register and login using c#, will create a new repo named `user-service`  (obviously). Planned features:
1. register without email verification
2. login, will generate JWT token
3. logout
4. verify-token, receive a JWT token returns user-id
5. change password
6. CRUD key-value. Any person can add new data to herself key-value pair, with string type for both

Tech wise:
1. grpc
2. C# dotnet 7
3. postgre
4. unusual coding style, will have context (with error in it) passed to every method and check if an error exist or not

any concerns, thoughts, comments?

**Comment by albilaga** [comment]
For number 4 will it be like we pass context (with error info) to next method? So if error, it will return early? Like [this](https://fsharpforfunandprofit.com/posts/recipe-part2/) ? So if we need to pass stack trace then we pass exception also?

**Reply by rizaramadan** [reply]
yeah, error information should added to the context to pin point where it occurs

**Reply by albilaga** [reply]
btw I just read [this](https://itnext.io/improving-code-readability-in-functional-c-using-linq-to-monad-d4c73194e9b1), and looks like we can use this also if we want
